File: Dataset/OOP_prepared/Python/Shape_2_trace_2.py
from math import pi

class Shape:
    def __init__(self, color="red", filled=True):
        self.color = color
        self.filled = filled

    def get_color(self):
        return self.color

    def set_color(self, color):
        self.color = color

    def is_filled(self):
        return self.filled

    def set_filled(self, filled):
        self.filled = filled

    def get_area(self):
        raise NotImplementedError("Subclass must implement abstract method")

    def get_perimeter(self):
        raise NotImplementedError("Subclass must implement abstract method")

    def __str__(self):
        return f"Shape[color={self.color}, filled={self.filled}]"


class Circle(Shape):
    def __init__(self, radius=1.0, color="red", filled=True):
        super().__init__(color, filled)
        self.radius = radius

    def get_area(self):
        return self.radius * self.radius * pi

    def get_perimeter(self):
        return 2 * self.radius * pi

    def __str__(self):
        return f"Circle[{super().__str__()}, radius={self.radius}]"

    def set_radius(self, radius):
        self.radius = radius


class Rectangle(Shape):
    def __init__(self, width=1.0, length=1.0, color="red", filled=True):
        super().__init__(color, filled)
        self.width = width
        self.length = length

    def get_width(self):
        return self.width

    def set_width(self, width):
        self.width = width

    def get_length(self):
        return self.length

    def set_length(self, length):
        self.length = length

    def get_area(self):
        return self.width * self.length

    def get_perimeter(self):
        return 2 * (self.width + self.length)

    def __str__(self):
        return f"Rectangle[{super().__str__()}, width={self.width}, length={self.length}]"


class Square(Rectangle):
    def __init__(self, side=1.0, color="red", filled=True):
        super().__init__(side, side, color, filled)

    def get_side(self):
        return self.get_length()

    def set_width(self, side):
        super().set_width(side)
        super().set_length(side)

    def set_length(self, side):
        super().set_width(side)
        super().set_length(side)

    def __str__(self):
        return f"Square[{super().__str__()}]"


if __name__ == "__main__":
    c2 = Circle(10)
    print(c2.get_area())

File: Dataset/Advanced/Test/OOP_prepared/Java/Shape_2_trace_2.java
package Shape;

public abstract class Shape {

    protected String color = "red";
    protected boolean filled = true;

    public Shape() {
    }

    public Shape(String color, boolean filled) {
        this.color = color;
        this.filled = filled;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public boolean isFilled() {
        return filled;
    }

    public void setFilled(boolean filled) {
        this.filled = filled;
    }

    public abstract double getArea();
    
    public abstract double getPerimeter();
    
    
    @Override
    public String toString() {
        return "Shape[" + "color=" + color + ",filled=" + filled + ']';
    }
}

public class Circle extends Shape{
    protected double radius=1.0;

    public Circle() {
    }
    
    public Circle(double radius) {
        this.radius = radius;
    }

    public Circle(double radius, String color, boolean filled) {
        super(color, filled);
        this.radius = radius;
    }
    
    public double getArea(){
        return this.radius*this.radius*Math.PI;
    }
    
    public double getPerimeter(){
        return this.radius*2*Math.PI;
    }

    @Override
    public String toString() {
        return "Circle["+ super.toString() + ",radius=" + radius + ']';
    }

    public void setRadius(int radius) {
         this.radius = radius;//To change body of generated methods, choose Tools | Templates.
    }
}


public class Rectangle extends Shape{
    protected double width;
    protected double length;

    public Rectangle() {
    }

    public Rectangle(double width, double length) {
        this.width = width;
        this.length = length;
    }

    public Rectangle(double width, double length, String color, boolean filled) {
        super(color, filled);
        this.width = width;
        this.length = length;
    }

    public double getWidth() {
        return width;
    }

    public void setWidth(double width) {
        this.width = width;
    }

    public double getLength() {
        return length;
    }

    public void setLength(double length) {
        this.length = length;
    }
    
    public double getArea(){
        return this.length*this.width;
    }
    
    public double getPerimeter(){
        return (this.length+this.width)*2;
    }

    @Override
    public String toString() {
        return "Rectangle["+ super.toString() + ",width=" + width + ", length=" + length + ']';
    }
    
    
}

public class Square extends Rectangle{

    public Square() {
    }

    public Square(double side) {
        super(side, side);
    }

    public Square(double side, String color, boolean filled) {
        super(side, side, color, filled);
    }
    
    public double getSide(){
        return this.getLength();
    }
    
    public void setWidth(double side){
        super.setWidth(side);
        super.setLength(side);
    }
    
    public void setLength(double side){
        super.setWidth(side);
        super.setLength(side);
    }
    
    @Override
    public String toString(){
        return "Square["+super.toString()+"]";
    }
}

public class TestMain {

    public static void main(String[] args) {
        Rectangle r1 = new Rectangle(4, 5, "green", false);
        System.out.println("Area: " + r1.getArea());
    }
}